edit-README

kodachi555 opened the pull request:
WHAT
データベースの設計内容をREADMEに記述
WHY
データベースの設計内容の確認のため

kodachi555 committed:
edit-README

append-databaseMarkdown

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
```
chatという仕様上、文字数制限のあるstring型にしましょう。また、写真もしくは文字があれば動く仕様なのでnull: false制約を外しましょう。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+|group_id|integer|null: false, foreign_key: true|
+
+### Association
+- belongs_to :groups
```
belongs_toにするなら単数形ですね。以下同様。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|name|text|null: false|
```
文字数制限のあるstring型にしましょう。また、後半に検索機能を付けるのでindexオプションを付けましょう。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+|------|----|-------|
+|name|text|null: false|
+|email|string|null: false, unique: true|
+|password|string|null: false|
```
passwordは自動生成されるので必要ありません。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+|name|text|null: false|
+|email|string|null: false, unique: true|
+|password|string|null: false|
+|message_id|integer|null: false, foreign_key: true|
```
usersのtableにmessage_idが必要か、再考してみましょう。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|name|text|null: false|
+|message_id|integer|null: false|
```
こちらも必要か、再考してみましょう。

MrUrakou commented on README.md:
```diff
@@ -22,3 +22,42 @@ Things you may want to cover:
+- has_many :groups :through => :messages
+
+
+## groupテーブル
```
groupsですね。

MrUrakou reviewed (commented):
groupsとusersのアソシエーションを今一度考えてみましょう。

kodachi555 committed:
correction-after-edit-README

WHAT
belongs_to single
WHY
belongできるのは単数

WHAT
user.name index: true
WHY
検索機能を実装予定

WHAT
## groupsテーブル
WHY
テーブルにレコード1つなのは稀

WHAT
削除：password
WHY
自動で追加されるため

WHAT
中間テーブルの追加とusers、groups間の中間テーブルを使用したアソシエーション追加

kodachi555 commented:
コミット名：correction-after-edit-README

WHAT
・ご指摘いただいた誤字訂正
・多対1の関係において、1の側から外部キーを削除
・多対多のテーブル間は中間テーブルで1対多対1にアソシエーションを組む

taketnk commented on README.md:
```diff
@@ -22,3 +22,55 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|body|string|null: false|
```
bodyがnull:falseだとimageだけ送信ができなくなってしまいますね。

kodachi555 replied:
訂正：
|body|string|null: false|
↓
|body|string|

taketnk commented on README.md:
```diff
@@ -22,3 +22,55 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|name|text|null: false|
```
group名はStringでもいいかと(256文字が超えるgroup名があることが予測されればtextでもいいです)

kodachi555 replied:
訂正：
|name|text|null: false|
↓
|name|string|null: false|

taketnk commented on README.md:
```diff
@@ -22,3 +22,55 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false|
+|group_id|integer|null: false|
```
外部キー制約をかけましょう。

kodachi555 replied:
訂正：
|user_id|integer|null: false|
|group_id|integer|null: false|
↓
|user_id|integer|null: false, foreign_key: true|
|group_id|integer|null: false, foreign_key: true|

kodachi555 committed:
correction-messages-body

訂正抜け
写真もしくは文字があれば動く仕様なのでnull: false制約を外しましょう。

kodachi555 committed:
correction

WHAT
groupNameType : text -> string
users_groupsOptions : foreign_key

kodachi555 committed:
correction comment

users_groupsのコメント部分の訂正：
末尾に「中間テーブル」と追記
